controllers/configurationpolicy_utils.go

```diff
@@ -55,6 +56,23 @@ func addRelatedObjects(
 	return relatedObjects
 }
 
+// unmarshalFromJSON unmarshals raw JSON data into an object
+func unmarshalFromJSON(rawData []byte) (unstructured.Unstructured, error) {
+	var unstruct unstructured.Unstructured
+	var blob interface{}
+
+	if jsonErr := json.Unmarshal(rawData, &blob); jsonErr != nil {
+		log.Error(jsonErr, "Could not unmarshal data from JSON")
+
+		return unstruct, jsonErr
+	}
+
+	unstruct.Object = make(map[string]interface{})
+	unstruct.Object = blob.(map[string]interface{})
+
+	return unstruct, nil
+}
+
```

Comment:
I created this new function since was using this flow a couple times and it seemed better to move it to the `utils` file.